.github/workflows/validate.yml

```diff
@@ -78,3 +78,78 @@ jobs:
     - name: Run agent tests
       run: .\scripts\runagenttests.bat
       shell: cmd
+  linux17:
+    runs-on: 'ubuntu-latest'
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-java@v1
+      with:
+        java-version: '17'
+        java-package: jdk
+    - name: Cache local Maven repository
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-17-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+    - name: Check formatting
+      run: ./scripts/checkformatting.sh
+    - name: Run core tests
+      run: ./scripts/runcoretests.sh
+    - name: Run application tests
+      run: ./scripts/runapptests.sh
+    - name: Run agent tests
+      run: ./scripts/runagenttests.sh
+
+  mac17:
+    runs-on: 'macOS-latest'
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-java@v1
+      with:
+        java-version: '17'
+        java-package: jdk
+    - name: Cache local Maven repository
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-17-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+    - name: Check formatting
+      run: ./scripts/checkformatting.sh
+    - name: Run core tests
+      run: ./scripts/runcoretests.sh
+    - name: Run application tests
+      run: ./scripts/runapptests.sh
+    - name: Run agent tests
+      run: ./scripts/runagenttests.sh
+
+  win17:
+    runs-on: 'windows-latest'
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-java@v1
+      with:
+        java-version: '17'
+        java-package: jdk
+    - name: Cache local Maven repository
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-17-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+    - name: Check formatting
+      run: .\scripts\checkformatting.bat
+      shell: cmd
+    - name: Run core tests
+      run: .\scripts\runcoretests.bat
+      shell: cmd
+    - name: Run application tests
+      run: .\scripts\runapptests.bat
+      shell: cmd
+    - name: Run agent tests
+      run: .\scripts\runagenttests.bat
+      shell: cmd
```

Comment:
This could be solved by define an according matrix definition and using `{{ matrix.jdk }}` in the respective places
```
    strategy:
      matrix:
        jdk: ['11', '17']
```
 Here the example for the `linux` build step:

```
jobs:
  linux:
    strategy:
      matrix:
        jdk: ['11', '17']
    runs-on: 'ubuntu-latest'
    steps:
    - uses: actions/checkout@v1
    - uses: actions/setup-java@v1
      with:
        java-version: {{ matrix.jdk }}
        java-package: jdk
    - name: Cache local Maven repository
      uses: actions/cache@v2
      with:
        path: ~/.m2/repository
        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
        restore-keys: |
          ${{ runner.os }}-maven-
    - name: Check formatting
      run: ./scripts/checkformatting.sh
    - name: Run core tests
      run: ./scripts/runcoretests.sh
    - name: Run application tests
      run: ./scripts/runapptests.sh
    - name: Run agent tests
      run: ./scripts/runagenttests.sh
```

Reply:
Thanks, I thought about the possibility, but forgot that it was that easy